src/BottomNavigation/BottomNavigation.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * TODOs:
+ * - Support custom highlight colors
+ * - Support custom highlight color per selected item
+ * - Support badged icons
+ * - Support showing label only for selected item
+ * - Add constraints for number of children provided
+ * - Enforce appropriate label length and avoid overflows/breaks
+ * - V2: Support custom background color on bar depending on selected item
+ * - V2: Support expanding width of selected item
+ * - V2: Add scrolling behaviours (hiding)
+ * - V2: Determine how to handle larger displays (menu on left)
+ */
+
+import React from 'react';
+
+function getStyles(props, context) {
+  const {
+    bottomNavigation,
+    zIndex,
+  } = context.muiTheme;
+
+  const styles = {
+    root: {
+      position: 'relative',
+      zIndex: zIndex.bottomNavigation,
+      width: '100%',
+      display: 'flex',
+      justifyContent: 'center',
+      backgroundColor: bottomNavigation.color,
+      height: bottomNavigation.height,
+    },
+    item: {
+      flex: '1',
+    },
+  };
+
+  return styles;
+}
+
+class BottomNavigation extends React.Component {
+  static muiName = 'BottomNavigation';
+
+  static propTypes = {
+    /**
+     * The `BottomNavigationItem`s to populate the `BottomNavigation` with.
+     */
+    children: React.PropTypes.node,
+
+    /**
+     * Applied to the bottom navigation's root element.
+     */
+    className: React.PropTypes.string,
+
+    /**
+     * The index of the currently selected navigation item.
+     */
+    selectedIndex: React.PropTypes.number,
+
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: React.PropTypes.object,
+  };
+
+  static contextTypes = {
+    muiTheme: React.PropTypes.object.isRequired,
+  };
+
+  render() {
+    const {
+      className,
+      children,
+      style,
+      selectedIndex,
+      ...other,
+    } = this.props;
+
+    const styles = getStyles(this.props, this.context);
+
+    const preparedChildren = React.Children.map(children, (child, index) => {
+      return React.cloneElement(child, {
```

Comment:
Can't we just clone the item that is active? We would save some memory allocation and CPU cycles.